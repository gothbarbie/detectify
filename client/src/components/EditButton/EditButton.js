import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import Icon from '../Icon'

const Button = styled.button`
  border: none;
  background-color: transparent;
`

const EditButton = ({ onClick }) => (
  <Button onClick={onClick}>
    <Icon icon={['fas', 'pencil-alt']} />
  </Button>
)

EditButton.propTypes = {
  onClick: PropTypes.func.isRequired,
}

export default EditButton
