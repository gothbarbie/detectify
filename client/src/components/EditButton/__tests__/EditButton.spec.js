import React from 'react'
import { shallow } from 'enzyme'

import EditButton from '../EditButton'

describe('Components/EditButton', () => {
  it('renders', () => {
    const props = {
      onClick: jest.fn(),
    }
    const wrapper = shallow(<EditButton {...props} />)

    expect(wrapper).toMatchSnapshot()
  })
})
