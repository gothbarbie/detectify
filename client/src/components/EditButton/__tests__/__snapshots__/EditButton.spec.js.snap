// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/EditButton renders 1`] = `
<styled.button
  onClick={[MockFunction]}
>
  <Icon
    icon={
      Array [
        "fas",
        "pencil-alt",
      ]
    }
  />
</styled.button>
`;
